spec/controllers/catalog_controller_spec.rb

```diff
@@ -215,6 +215,24 @@
       get :index, params: { q: "" }
       expect(response).to be_success
     end
+
+    # rubocop:disable RSpec/AnyInstance
+    context "in read-only mode" do
+      let(:emsg) { "PG::InsufficientPrivilege: ERROR:  permission denied for relation searches" }
+      before do
+        allow(Figgy).to receive(:read_only_mode).and_return(true)
+        allow_any_instance_of(Blacklight::SearchContext).to receive(:find_or_initialize_search_session_from_params).and_raise ActiveRecord::StatementInvalid, emsg
```

Comment:
Is there a way to do this without `allow_any_instance`, or is there to too much mocking involved? FWIW, I was just poking around blacklight and was astounded by how many levels of nested concerns there are in the controller.